Examples fail to compile

I tried debugging an issue with `test-fuzz` and wanted to check if I could do the bare minimum of compiling the examples. They do not work. 

# Steps to reproduce

1. `git clone git@github.com:trailofbits/test-fuzz.git`
2. `cd test-fuzz/examples`
3. `cargo install cargo-test-fuzz afl`
4. `cargo test`

Failure at this point. 

# Expected 

Able to instrument and run `cargo test-fuzz`. 

# Actual

```
❰app❙~/Git/test-fuzz/examples(git:master)❱✔≻ cargo test                                               (master) 14:38
warning: unreachable definition
--> macro/src/lib.rs:409:9
|
408 |     let serde_format = serde_format();
|                        -------------- any code following this expression is unreachable
409 |     let write_concretizations = quote! {
|         ^^^^^^^^^^^^^^^^^^^^^ unreachable definition
|
= note: `#[warn(unreachable_code)]` on by default
note: this expression has type `SerdeFormat`, which is uninhabited
--> macro/src/lib.rs:408:24
|
408 |     let serde_format = serde_format();
|                        ^^^^^^^^^^^^^^

warning: unused variable: `stmts`
--> macro/src/lib.rs:225:9
|
225 |     let stmts = &block.stmts;
|         ^^^^^ help: if this is intentional, prefix it with an underscore: `_stmts`
|
= note: `#[warn(unused_variables)]` on by default

warning: unused variable: `combined_type_idents`
--> macro/src/lib.rs:294:9
|
294 |     let combined_type_idents = [impl_ty_idents.clone(), ty_idents.clone()].concat();
|         ^^^^^^^^^^^^^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_combined_type_idents`

warning: unused variable: `impl_ty_names`
--> macro/src/lib.rs:296:9
|
296 |     let impl_ty_names: Vec<Expr> = impl_ty_idents
|         ^^^^^^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_impl_ty_names`

warning: unused variable: `ty_names`
--> macro/src/lib.rs:300:9
|
300 |     let ty_names: Vec<Expr> = ty_idents
|         ^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_ty_names`

warning: unused variable: `impl_generics`
--> macro/src/lib.rs:308:10
|
308 |     let (impl_generics, ty_generics, where_clause) = combined_generics.split_for_impl();
|          ^^^^^^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_impl_generics`

warning: unused variable: `where_clause`
--> macro/src/lib.rs:308:38
|
308 |     let (impl_generics, ty_generics, where_clause) = combined_generics.split_for_impl();
|                                      ^^^^^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_where_clause`

warning: unused variable: `impl_generics_deserializable`
--> macro/src/lib.rs:309:10
|
309 |     let (impl_generics_deserializable, _, _) = combined_generics_deserializable.split_for_impl();
|          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_impl_generics_deserializable`

warning: unused variable: `impl_concretization`
--> macro/src/lib.rs:335:9
|
335 |     let impl_concretization = opts_concretize_impl.as_ref().map(args_as_turbofish);
|         ^^^^^^^^^^^^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_impl_concretization`

warning: unused variable: `concretization`
--> macro/src/lib.rs:336:9
|
336 |     let concretization = opts_concretize.as_ref().map(args_as_turbofish);
|         ^^^^^^^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_concretization`

warning: unused variable: `combined_concretization`
--> macro/src/lib.rs:337:9
|
337 |     let combined_concretization =
|         ^^^^^^^^^^^^^^^^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_combined_concretization`

warning: unused variable: `self_ty_base`
--> macro/src/lib.rs:345:9
|
345 |     let self_ty_base = self_ty.as_ref().map(type_utils::type_base);
|         ^^^^^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_self_ty_base`

warning: unused variable: `receiver`
--> macro/src/lib.rs:347:10
|
347 |     let (receiver, mut arg_tys, fmt_args, mut ser_args, de_args) =
|          ^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_receiver`

warning: unused variable: `fmt_args`
--> macro/src/lib.rs:347:33
|
347 |     let (receiver, mut arg_tys, fmt_args, mut ser_args, de_args) =
|                                 ^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_fmt_args`

warning: unused variable: `de_args`
--> macro/src/lib.rs:347:57
|
347 |     let (receiver, mut arg_tys, fmt_args, mut ser_args, de_args) =
|                                                         ^^^^^^^ help: if this is intentional, prefix it with an underscore: `_de_args`

warning: unused variable: `args_is`
--> macro/src/lib.rs:352:9
|
352 |     let args_is: Vec<Expr> = arg_tys
|         ^^^^^^^ help: if this is intentional, prefix it with an underscore: `_args_is`

warning: unused variable: `args_from_autos`
--> macro/src/lib.rs:368:9
|
368 |     let args_from_autos = args_from_autos(&autos);
|         ^^^^^^^^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_args_from_autos`

warning: unused variable: `ret_ty`
--> macro/src/lib.rs:369:9
|
369 |     let ret_ty = match &sig.output {
|         ^^^^^^ help: if this is intentional, prefix it with an underscore: `_ret_ty`

warning: unused variable: `mod_ident`
--> macro/src/lib.rs:378:9
|
378 |     let mod_ident = Ident::new(&format!("{}_fuzz", renamed_target_ident), Span::call_site());
|         ^^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_mod_ident`

warning: unused variable: `ty_generics_as_turbofish`
--> macro/src/lib.rs:390:10
|
390 |     let (ty_generics_as_turbofish, struct_args) = if opts.only_concretizations {
|          ^^^^^^^^^^^^^^^^^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_ty_generics_as_turbofish`

warning: unused variable: `struct_args`
--> macro/src/lib.rs:390:36
|
390 |     let (ty_generics_as_turbofish, struct_args) = if opts.only_concretizations {
|                                    ^^^^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_struct_args`

warning: unused variable: `serde_format`
--> macro/src/lib.rs:408:9
|
408 |     let serde_format = serde_format();
|         ^^^^^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_serde_format`

warning: unused variable: `attrs`
--> macro/src/lib.rs:219:5
|
219 |     attrs: &Vec<Attribute>,
|     ^^^^^ help: if this is intentional, prefix it with an underscore: `_attrs`

warning: unused variable: `vis`
--> macro/src/lib.rs:220:5
|
220 |     vis: &Visibility,
|     ^^^ help: if this is intentional, prefix it with an underscore: `_vis`

warning: unused variable: `defaultness`
--> macro/src/lib.rs:221:5
|
221 |     defaultness: &Option<token::Default>,
|     ^^^^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_defaultness`

warning: unused variable: `reader`
--> runtime/src/lib.rs:194:75
|
194 | pub fn read_args<T: DeserializeOwned, R: Read>(serde_format: SerdeFormat, reader: R) -> Option<T> {
|                                                                           ^^^^^^ help: if this is intentional,prefix it with an underscore: `_reader`
|
= note: `#[warn(unused_variables)]` on by default

warning: `test-fuzz-macro` (lib) generated 25 warnings
warning: `test-fuzz-runtime` (lib) generated 1 warning
Compiling test-fuzz-examples v2.0.0 (/mnt/Archive/Git/test-fuzz/examples)
error: custom attribute panicked
--> examples/tests/rename.rs:1:1
|
1 | #[test_fuzz::test_fuzz(rename = "bar")]
| ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
|
= help: message: No serde format selected

error: custom attribute panicked
--> examples/tests/parse_duration.rs:7:1
|
7 | #[test_fuzz::test_fuzz]
| ^^^^^^^^^^^^^^^^^^^^^^^
|
= help: message: No serde format selected

error: custom attribute panicked
--> examples/tests/debug.rs:14:5
|
14 |     #[test_fuzz::test_fuzz]
|     ^^^^^^^^^^^^^^^^^^^^^^^
|
= help: message: No serde format selected

error: custom attribute panicked
--> examples/tests/default.rs:7:5
|
7 |     #[test_fuzz::test_fuzz]
|     ^^^^^^^^^^^^^^^^^^^^^^^
|
= help: message: No serde format selected

error: custom attribute panicked
--> examples/tests/arc.rs:23:1
|
23 | #[test_fuzz::test_fuzz]
| ^^^^^^^^^^^^^^^^^^^^^^^
|
= help: message: No serde format selected

error: custom attribute panicked
--> examples/tests/assert.rs:1:1
|
1 | #[test_fuzz::test_fuzz]
| ^^^^^^^^^^^^^^^^^^^^^^^
|
= help: message: No serde format selected

error: custom attribute panicked
--> examples/tests/unserde.rs:4:1
|
4 | #[test_fuzz::test_fuzz(only_concretizations)]
| ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
|
= help: message: No serde format selected

error[E0425]: cannot find function `parse` in this scope
--> examples/tests/parse_duration.rs:14:5
|
14 |     parse("1e5 ns").unwrap();
|     ^^^^^ not found in this scope
|
help: consider importing this function
|
4  | use parse_duration::parse;
|

warning: unused import: `parse_duration::parse::Error`
--> examples/tests/parse_duration.rs:4:5
|
4 | use parse_duration::parse::Error;
|     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
|
= note: `#[warn(unused_imports)]` on by default

warning: unused import: `std::time::Duration`
--> examples/tests/parse_duration.rs:5:5
|
5 | use std::time::Duration;
|     ^^^^^^^^^^^^^^^^^^^

error: custom attribute panicked
--> examples/tests/debug.rs:32:5
|
32 |     #[test_fuzz::test_fuzz]
|     ^^^^^^^^^^^^^^^^^^^^^^^
|
= help: message: No serde format selected

error[E0425]: cannot find function `foo` in this scope
--> examples/tests/rename.rs:10:5
|
10 |     foo();
|     ^^^ not found in this scope

error: custom attribute panicked
--> examples/tests/default.rs:22:5
|
22 |     #[test_fuzz::test_fuzz]
|     ^^^^^^^^^^^^^^^^^^^^^^^
|
= help: message: No serde format selected

error: custom attribute panicked
--> examples/tests/unserde.rs:7:1
|
7 | #[test_fuzz::test_fuzz(enable_in_production, only_concretizations)]
| ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
|
= help: message: No serde format selected

error[E0425]: cannot find function `target` in this scope
--> examples/tests/arc.rs:18:13
|
18 |             target(arc, i);
|             ^^^^^^ not found in this scope

error: custom attribute panicked
--> examples/tests/from.rs:18:1
|
18 | #[test_fuzz::test_fuzz_impl]
| ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
|
= help: message: No serde format selected

warning: unused import: `current`
--> examples/tests/arc.rs:3:14
|
3 |     thread::{current, spawn},
|              ^^^^^^^
|
= note: `#[warn(unused_imports)]` on by default

error: custom attribute panicked
--> examples/tests/lifetime.rs:26:1
|
26 | #[test_fuzz::test_fuzz]
| ^^^^^^^^^^^^^^^^^^^^^^^
|
= help: message: No serde format selected

For more information about this error, try `rustc --explain E0425`.
error: custom attribute panicked
--> examples/src/main.rs:5:1
|
5 | #[test_fuzz::test_fuzz(enable_in_production)]
| ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
|
= help: message: No serde format selected

error: custom attribute panicked
--> examples/tests/alloc.rs:1:1
|
1 | #[test_fuzz::test_fuzz(no_auto_generate)]
| ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
|
= help: message: No serde format selected

error: custom attribute panicked
--> examples/tests/from.rs:26:1
|
26 | #[test_fuzz::test_fuzz_impl]
| ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
|
= help: message: No serde format selected

warning: `test-fuzz-examples` (test "parse_duration") generated 2 warnings
error: could not compile `test-fuzz-examples` due to 2 previous errors; 2 warnings emitted
warning: build failed, waiting for other jobs to finish...
error[E0425]: cannot find function `target` in this scope
--> examples/tests/assert.rs:8:5
|
8 |     target(false);
|     ^^^^^^ not found in this scope

error[E0425]: cannot find function `target` in this scope
--> examples/tests/default.rs:12:9
|
12 |         target(&Struct(true));
|         ^^^^^^ not found in this scope

error[E0425]: cannot find function `target` in this scope
--> examples/tests/default.rs:27:9
|
27 |         target(&Struct(true));
|         ^^^^^^ not found in this scope

error: custom attribute panicked
--> examples/tests/from.rs:35:1
|
35 | #[test_fuzz::test_fuzz_impl]
| ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
|
= help: message: No serde format selected

error[E0425]: cannot find function `target` in this scope
--> examples/tests/lifetime.rs:31:5
|
31 |     target(Bar(&Foo));
|     ^^^^^^ not found in this scope

warning: unused import: `TryFrom`
--> examples/tests/from.rs:7:20
|
7 | use std::convert::{TryFrom, TryInto};
|                    ^^^^^^^
|
= note: `#[warn(unused_imports)]` on by default

error[E0425]: cannot find function `target` in this scope
--> examples/src/main.rs:2:5
|
2 |     target("Hello, world!");
|     ^^^^^^ not found in this scope

error[E0425]: cannot find function `target` in this scope
--> examples/tests/unserde.rs:12:5
|
12 |     target(&Struct);
|     ^^^^^^ not found in this scope

error[E0425]: cannot find function `target_in_production` in this scope
--> examples/tests/unserde.rs:17:5
|
17 |     target_in_production(&Struct);
|     ^^^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find function `target` in this scope
--> examples/tests/alloc.rs:10:5
|
10 |     target(0);
|     ^^^^^^ not found in this scope

warning: `test-fuzz-examples` (test "arc") generated 1 warning
error: custom attribute panicked
--> examples/tests/serde.rs:13:5
|
13 |     #[test_fuzz::test_fuzz(no_auto_generate)]
|     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
|
= help: message: No serde format selected

error: custom attribute panicked
--> examples/tests/serde.rs:68:5
|
68 |     #[test_fuzz::test_fuzz]
|     ^^^^^^^^^^^^^^^^^^^^^^^
|
= help: message: No serde format selected

error: custom attribute panicked
--> examples/tests/serde.rs:88:5
|
88 |     #[test_fuzz::test_fuzz]
|     ^^^^^^^^^^^^^^^^^^^^^^^
|
= help: message: No serde format selected

error: custom attribute panicked
--> examples/tests/serde.rs:128:5
|
128 |     #[test_fuzz::test_fuzz]
|     ^^^^^^^^^^^^^^^^^^^^^^^
|
= help: message: No serde format selected

error: custom attribute panicked
--> examples/tests/serde.rs:146:5
|
146 |     #[test_fuzz::test_fuzz]
|     ^^^^^^^^^^^^^^^^^^^^^^^
|
= help: message: No serde format selected

error: custom attribute panicked
--> examples/tests/serde.rs:164:5
|
164 |     #[test_fuzz::test_fuzz]
|     ^^^^^^^^^^^^^^^^^^^^^^^
|
= help: message: No serde format selected

error: custom attribute panicked
--> examples/tests/serde.rs:193:5
|
193 |     #[test_fuzz::test_fuzz]
|     ^^^^^^^^^^^^^^^^^^^^^^^
|
= help: message: No serde format selected

error: custom attribute panicked
--> examples/tests/serde.rs:215:5
|
215 |     #[test_fuzz::test_fuzz]
|     ^^^^^^^^^^^^^^^^^^^^^^^
|
= help: message: No serde format selected

error: custom attribute panicked
--> examples/tests/serde.rs:249:5
|
249 |     #[test_fuzz::test_fuzz]
|     ^^^^^^^^^^^^^^^^^^^^^^^
|
= help: message: No serde format selected

error[E0277]: the trait bound `Bar: From<Foo>` is not satisfied
--> examples/tests/from.rs:45:24
|
45 |     let bar: Bar = Foo.into();
|                        ^^^^ the trait `From<Foo>` is not implemented for `Bar`
|
= note: required because of the requirements on the impl of `std::convert::Into<Bar>` for `Foo`

error[E0277]: the trait bound `Baz: From<Bar>` is not satisfied
--> examples/tests/from.rs:46:24
|
46 |     let baz: Baz = bar.try_into().unwrap();
|                        ^^^^^^^^ the trait `From<Bar>` is not implemented for `Baz`
|
= note: required because of the requirements on the impl of `std::convert::Into<Baz>` for `Bar`
= note: required because of the requirements on the impl of `TryFrom<Bar>` for `Baz`
= note: required because of the requirements on the impl of `TryInto<Baz>` for `Bar`

error[E0277]: the trait bound `Foo: From<Baz>` is not satisfied
--> examples/tests/from.rs:47:22
|
47 |     let _: Foo = baz.into();
|                      ^^^^ the trait `From<Baz>` is not implemented for `Foo`
|
= note: required because of the requirements on the impl of `std::convert::Into<Foo>` for `Baz`

For more information about this error, try `rustc --explain E0277`.
warning: `test-fuzz-examples` (test "from") generated 1 warning
error: custom attribute panicked
--> examples/tests/serde.rs:272:5
|
272 |     #[test_fuzz::test_fuzz]
|     ^^^^^^^^^^^^^^^^^^^^^^^
|
= help: message: No serde format selected

error: custom attribute panicked
--> examples/tests/serde.rs:306:5
|
306 |     #[test_fuzz::test_fuzz]
|     ^^^^^^^^^^^^^^^^^^^^^^^
|
= help: message: No serde format selected

error: custom attribute panicked
--> examples/tests/serde.rs:341:5
|
341 |     #[test_fuzz::test_fuzz]
|     ^^^^^^^^^^^^^^^^^^^^^^^
|
= help: message: No serde format selected

error: custom attribute panicked
--> examples/tests/serde.rs:375:5
|
375 |     #[test_fuzz::test_fuzz]
|     ^^^^^^^^^^^^^^^^^^^^^^^
|
= help: message: No serde format selected

error: custom attribute panicked
--> examples/tests/serde.rs:398:5
|
398 |     #[test_fuzz::test_fuzz]
|     ^^^^^^^^^^^^^^^^^^^^^^^
|
= help: message: No serde format selected

error: custom attribute panicked
--> examples/tests/serde.rs:428:5
|
428 |     #[test_fuzz::test_fuzz]
|     ^^^^^^^^^^^^^^^^^^^^^^^
|
= help: message: No serde format selected

error: custom attribute panicked
--> examples/tests/serde.rs:462:5
|
462 |     #[test_fuzz::test_fuzz]
|     ^^^^^^^^^^^^^^^^^^^^^^^
|
= help: message: No serde format selected

error[E0425]: cannot find function `target` in this scope
--> examples/tests/serde.rs:48:9
|
48 |         target(
|         ^^^^^^ not found in this scope

error[E0425]: cannot find function `target` in this scope
--> examples/tests/serde.rs:78:9
|
78 |         target(
|         ^^^^^^ not found in this scope

error[E0425]: cannot find function `target` in this scope
--> examples/tests/serde.rs:113:9
|
113 |         target(
|         ^^^^^^ not found in this scope

error[E0425]: cannot find function `target` in this scope
--> examples/tests/serde.rs:137:9
|
137 |         target(
|         ^^^^^^ not found in this scope

error[E0425]: cannot find function `target` in this scope
--> examples/tests/serde.rs:155:9
|
155 |         target(<()>::default(), &<()>::default(), &mut <()>::default());
|         ^^^^^^ not found in this scope

error[E0425]: cannot find function `target` in this scope
--> examples/tests/serde.rs:177:9
|
177 |         target(
|         ^^^^^^ not found in this scope

error[E0425]: cannot find function `target` in this scope
--> examples/tests/serde.rs:206:9
|
206 |         target(UnitVariant::A, &UnitVariant::B, &mut UnitVariant::B);
|         ^^^^^^ not found in this scope

error[E0425]: cannot find function `target` in this scope
--> examples/tests/serde.rs:228:9
|
228 |         target(
|         ^^^^^^ not found in this scope

error[E0425]: cannot find function `target` in this scope
--> examples/tests/serde.rs:262:9
|
262 |         target(
|         ^^^^^^ not found in this scope

error[E0425]: cannot find function `target` in this scope
--> examples/tests/serde.rs:293:9
|
293 |         target(
|         ^^^^^^ not found in this scope

error[E0425]: cannot find function `target` in this scope
--> examples/tests/serde.rs:325:9
|
325 |         target(
|         ^^^^^^ not found in this scope

error[E0425]: cannot find function `target` in this scope
--> examples/tests/serde.rs:354:9
|
354 |         target(
|         ^^^^^^ not found in this scope

error[E0425]: cannot find function `target` in this scope
--> examples/tests/serde.rs:388:9
|
388 |         target(
|         ^^^^^^ not found in this scope

error[E0425]: cannot find function `target` in this scope
--> examples/tests/serde.rs:411:9
|
411 |         target(
|         ^^^^^^ not found in this scope

error[E0425]: cannot find function `target` in this scope
--> examples/tests/serde.rs:437:9
|
437 |         target(
|         ^^^^^^ not found in this scope

error[E0425]: cannot find function `target` in this scope
--> examples/tests/serde.rs:475:9
|
475 |         target(
|         ^^^^^^ not found in this scope

error: build failed
❰app❙~/Git/test-fuzz/examples(git:master)❱✘≻                                                  0.805s  (master) 14:38
```

--- a/examples/Cargo.toml
+++ b/examples/Cargo.toml
@@ -10,7 +10,7 @@ path = "src/main.rs"
 
 [dependencies]
 serde = { version = "1.0.132", features = ["rc"] }
-test-fuzz = { path = "../test-fuzz", version = "=2.0.0", default-features = false }
+test-fuzz = { path = "../test-fuzz", version = "=2.0.0" }
 
 [dev-dependencies]
 lazy_static = "1.4.0"

